diff --git a/dwm.c b/dwm.c
index 17355e1..087d3dd 100644
--- a/dwm.c
+++ b/dwm.c
@@ -183,6 +183,7 @@ static void configure(Client *c);
 static void configurenotify(XEvent *e);
 static void configurerequest(XEvent *e);
 static Monitor *createmon(void);
+static void cycle(const Arg *arg);
 static void destroynotify(XEvent *e);
 static void detach(Client *c);
 static void detachstack(Client *c);
@@ -233,11 +234,13 @@ static void setfullscreen(Client *c, int fullscreen);
 static void setlayout(const Arg *arg);
 static void setmfact(const Arg *arg);
 static void setup(void);
+static int shifttag(int dist);
 static void showhide(Client *c);
 static void sigchld(int unused);
 static void spawn(const Arg *arg);
 static Monitor *systraytomon(Monitor *m);
 static void tag(const Arg *arg);
+static void tagcycle(const Arg *arg);
 static void tagmon(const Arg *arg);
 static void tile(Monitor *);
 static void togglebar(const Arg *arg);
@@ -743,6 +746,13 @@ createmon(void)
 }
 
 void
+cycle(const Arg *arg) {
+	const Arg a = { .i = shifttag(arg->i) };
+
+	view(&a);
+}
+
+void
 destroynotify(XEvent *e)
 {
 	Client *c;
@@ -758,6 +768,14 @@ destroynotify(XEvent *e)
 }
 
 void
+tagcycle(const Arg *arg) {
+	const Arg a = { .i = shifttag(arg->i) };
+
+	tag(&a);
+	view(&a);
+}
+
+void
 detach(Client *c)
 {
 	Client **tc;
@@ -1788,6 +1806,17 @@ setup(void)
 	focus(NULL);
 }
 
+int
+shifttag(int dist) {
+	int seltags = selmon->tagset[selmon->seltags] & TAGMASK;
+
+	if(dist > 0) /* left circular shift */
+		seltags = (seltags << dist) | (seltags >> (LENGTH(tags) - dist));
+	else /* right circular shift */
+		seltags = (seltags >> (- dist)) | (seltags << (LENGTH(tags) + dist));
+	return seltags;
+}
+
 void
 showhide(Client *c)
 {
